File: backend/src/main/java/com/novus/backend/domain/content/Quote.java
package com.novus.backend.domain.content;

import com.novus.backend.common.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "quotes")
@Getter
@Setter
public class Quote extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String content;

    private String author;

    private String category;

    @Column(name = "is_premium")
    private boolean isPremium = false;

    @Column(name = "is_religious")
    private boolean isReligious = false;

    @Column(name = "background_image_url", columnDefinition = "TEXT")
    private String backgroundImageUrl;
}

File: mobile/services/quote.ts
import api from './api';

export interface Quote {
    id: number;
    content: string;
    author?: string;
    category?: string;
    backgroundImageUrl?: string;
    isPremium: boolean;
    isReligious: boolean;
}

export const quoteService = {
    getRandomQuotes: async (limit: number = 10): Promise<Quote[]> => {
        const response = await api.get<{ data: Quote[] }>(`/quotes/random?limit=${limit}`);
        return response.data.data;
    },

    getQuotesByAuthor: async (author: string): Promise<Quote[]> => {
        const response = await api.get<{ data: Quote[] }>(`/quotes?author=${encodeURIComponent(author)}`);
        return response.data.data;
    },

    getQuoteById: async (id: number): Promise<Quote> => {
        const response = await api.get<{ data: Quote }>(`/quotes/${id}`);
        return response.data.data;
    },
};
